openshift/openshift_transport.go

```diff
@@ -44,67 +43,32 @@ func (t openshiftTransport) ValidatePolicyConfigurationScope(scope string) error
 
 // openshiftReference is an ImageReference for OpenShift images.
 type openshiftReference struct {
-	baseURL         *url.URL
 	namespace       string
 	stream          string
-	tag             string
-	dockerReference reference.Named // Computed from the above in advance, so that later references can not fail.
+	dockerReference reference.NamedTagged // Computed from the above in advance, so that later references can not fail.
 }
 
-// FIXME: Is imageName like this a good way to refer to OpenShift images?
-// Keep this in sync with scopeRegexp!
-var imageNameRegexp = regexp.MustCompile("^([^:/]*)/([^:/]*):([^:/]*)$")
-
 // ParseReference converts a string, which should not start with the ImageTransport.Name prefix, into an OpenShift ImageReference.
-func ParseReference(reference string) (types.ImageReference, error) {
-	// Overall, this is modelled on openshift/origin/pkg/cmd/util/clientcmd.New().ClientConfig() and openshift/origin/pkg/client.
-	cmdConfig := defaultClientConfig()
-	logrus.Debugf("cmdConfig: %#v", cmdConfig)
-	restConfig, err := cmdConfig.ClientConfig()
-	if err != nil {
-		return nil, err
-	}
-	// REMOVED: SetOpenShiftDefaults (values are not overridable in config files, so hard-coded these defaults.)
-	logrus.Debugf("restConfig: %#v", restConfig)
-	baseURL, _, err := restClientFor(restConfig)
-	if err != nil {
-		return nil, err
-	}
-	logrus.Debugf("URL: %#v", *baseURL)
-
-	m := imageNameRegexp.FindStringSubmatch(reference)
-	if m == nil || len(m) != 4 {
-		return nil, fmt.Errorf("Invalid image reference %s, %#v", reference, m)
+func ParseReference(ref string) (types.ImageReference, error) {
+	r, err := reference.ParseNamed(ref)
+	// I believe it's an hard requirement to be namespace/stream:tag right?
+	tagged, ok := r.(reference.NamedTagged)
+	if err != nil && !ok {
```

Comment:
This should probably be `||`; and handling the two separately would probably allow a bit better error messages.